torch/distributed/_tensor/device_mesh.py

```diff
@@ -269,3 +272,68 @@ def get_coordinate(self) -> Optional[List[int]]:
         dimensions of the mesh. If this rank is not part of the mesh, return None.
         """
         return self._coordinate_on_dim if self._coordinate_on_dim else None
+
+
+@no_type_check
+def init_device_mesh(
+    device_type: str,
+    mesh_dims: Optional[Sequence[int]] = None,
+    mesh_dim_names: Optional[Sequence[str]] = None,
```

Comment:
for both mesh_shape and mesh_dim_names, we should not taking sequence of ints, we should take tuple of ints, this is to disable user from modifying the list after calls this API